### ISSUE TITLE
"showing" gradient mask broken in dt 3.1.0+222

### ISSUE THREAD

**Issue by jenshannoschwalm** [darktable-org/darktable#3907]
While

- defining a new gradient mask or 
- using a developed image and use the "show and edit mask elements"

you will observe these problems (the mask visualizing and the desired effect seem to be ok)

1. the gradient is displayed completely wrong like in this
![Bildschirmfoto von 2019-12-28 21-40-36](https://user-images.githubusercontent.com/50982232/71549326-fd173080-29bb-11ea-8f97-c907ab186331.png)

2. At the same time dt is **very** unresponsive and cpu consumption is ~100% for one thread, this takes up to a few seconds (cairo seems to be drawing ...)
3. you will not be able to see the "mask list" as here, .
![Bildschirmfoto von 2019-12-28 21-43-02](https://user-images.githubusercontent.com/50982232/71549384-afe78e80-29bc-11ea-8bb7-145128743a0a.png)

Conditions to trigger
1. Use a fresh edit on any image, create a gradient and position it. Or "show" the gradient mask in an already developed image 
2. lens correction is a pre-requisite, problem is **only** detectable if lens correction is on
3. Cl or cpu code doesn't matter

 - Darktable Version: current git master (compiled after clearing build and removing all cl kernels)
 - OS: Fedora
 - nvidia 440.44

**Comment by TurboGit** [comment]
Really strange... There must be something more as I cannot reproduce !

**Comment by upegelow** [comment]
Am 29.12.19 um 08:50 schrieb Pascal Obry:
> Really strange... There must be something more as I cannot reproduce !

I think I've spotted the problem. A bug I have introduced with curved 
gradients. Will fix asap.

**Comment by jenshannoschwalm** [comment]
@upegelow thanks, i just wanted to ping you.

Not being able to reproduce might be related to
1. You **must** have the lens correction activated **before** switching to the module where you view the mask.
2. The problem is significant only if the lens correction is pretty strong (try LeicaQ test images for example)

The wrong lines & "cairo doing wrong stuff" are visible in masks/gradient.c ~L.615. I tested with
```
    {
      _gradient_point_transform(xref, yref, gpt->points[i * 2] + dx, gpt->points[i * 2 + 1] + dy, sinv, cosv,
                                &x, &y);
      fprintf(stderr," (%.1f|%.1f)",x,y);
      cairo_line_to(cr, x, y);
    }
```
and many x/y are nan (cairo takes long in these cases...)

**Comment by upegelow** [comment]
Am 29.12.19 um 17:42 schrieb Hanno Schwalm:
> @upegelow <https://github.com/upegelow> thanks, i just wanted to ping you.
> 

Root cause has been a silly bug (divide by zero). Should be fixed by #3929.

**Comment by jenshannoschwalm** [comment]
Fixed by #3929
